skrub/_dataframe/_common.py

```diff
@@ -325,25 +325,32 @@ def _all_null_like_polars(col, length=None, dtype=None, name=None):
 
 
 @dispatch
-def concat_horizontal(*dataframes):
+def concat(*dataframes, axis=0):
     raise NotImplementedError()
 
 
-@concat_horizontal.specialize("pandas")
-def _concat_horizontal_pandas(*dataframes):
-    init_index = dataframes[0].index
-    dataframes = [df.reset_index(drop=True) for df in dataframes]
-    dataframes = _join_utils.make_column_names_unique(*dataframes)
+@concat.specialize("pandas")
+def _concat_pandas(*dataframes, axis=0):
     kwargs = {"copy": False} if pandas_version < parse_version("3.0") else {}
-    result = pd.concat(dataframes, axis=1, **kwargs)
-    result.index = init_index
-    return result
-
-
-@concat_horizontal.specialize("polars")
-def _concat_horizontal_polars(*dataframes):
-    dataframes = _join_utils.make_column_names_unique(*dataframes)
-    return pl.concat(dataframes, how="horizontal")
+    if axis in [0, "index"]:
+        return pd.concat(dataframes, axis=0, **kwargs)
+    else:
+        init_index = dataframes[0].index
+        dataframes = [df.reset_index(drop=True) for df in dataframes]
+        dataframes = _join_utils.make_column_names_unique(*dataframes)
+        kwargs = {"copy": False} if pandas_version < parse_version("3.0") else {}
+        result = pd.concat(dataframes, axis=1, **kwargs)
+        result.index = init_index
+        return result
+
+
+@concat.specialize("polars")
+def _concat_polars(*dataframes, axis=0):
+    if axis in [0, "index"]:
+        return pl.concat(dataframes, how="vertical")
```

Comment:
If we want `_concat_polars` and `_concat_pandas` to behave the same, we should use:
```suggestion
        return pl.concat(dataframes, how="diagonal_relaxed")
```
- "diagonal" allows dataframes with different column names
- "relaxed" allows dataframes whose matching columns have different data type